impl/src/main/java/org/glassfish/soteria/identitystores/LdapIdentityStore.java

```diff
@@ -218,11 +228,24 @@ private CredentialValidationResult checkDirectBinding(UsernamePasswordCredential
         );
 
         // If this doesn't throw an exception internally, the caller dn exists and the password is correct
-        LdapContext ldapContext = createLdapContext(
+        LdapContext ldapContext = null;
+        try {
+          ldapContext = createLdapContext(
                 ldapIdentityStoreDefinition.url(),
                 callerDn,
                 new String(usernamePasswordCredential.getPassword().getValue())
-        );
+          );
+        } catch (Exception e) {
```

Comment:
Instead of doing `catch(Exception)` and then using `instanceof` to handle different ones, it's probably better to `catch` each exception that you want to handle, and let the rest propagate (or swallow them).